### PR TITLE
Addressing #3805: allow iteration over columns

### LINKED ISSUE #3805: Add easy way to iterate over table columns
I often find myself doing:

```
for colname in t.columns:
    column = t[colname]
```

It would be nice to provide a way to iterate over columns, for example:

```
for column in t.itercols():
    # ta-da
```

@taldcroft - what do you think?

Now ideally we'd have used ``t.columns`` for that since at the moment ``t.colnames`` and ``t.columns`` seem to do the same thing, but I don't think we want to break backward-compatibility there.

### PR DESCRIPTION
**Pull request by pulsestaysconstant**
Created an itercols method to provide a simpler way to iterate over columns in a Table. Added a test for itercols, as well. 

edit: Closes #3805

### CODE REVIEW

**Comment by embray** [comment]
Not my preference--compared to making `Table.columns` list-like--but it does the thing and does it right.  Please just add a docstring, and a changelog entry under the v1.2->New Features->astropy.table section.  Thanks! :+1:

**Comment by mwcraig** [comment]
@pulsestaysconstant -- could you please rebase this?

File: astropy/table/table.py [hunk 1]
```diff
@@ -737,6 +737,14 @@ def _make_table_from_cols(table, cols):
 
         table.columns = columns
 
+    def itercols(self):
+        """
+        Allows for simpler iteration over columns,
```
Comment (taldcroft) [inline astropy/table/table.py]: The docstring should say directly what this does.  So maybe:
```
Iterate over the columns of this table.

Example
-----------
t = Table([[1], [2]])
for col in t.itercols():
    print(col)

Using ``itercols()`` is similar to  ``for col in t.columns.values()`` 
but is syntactically preferred.
```

File: astropy/table/tests/test_table.py [hunk 2]
```diff
@@ -333,6 +333,14 @@ def test_2(self, table_types):
         with pytest.raises(KeyError):
             t['b']  # column does not exist
 
+    def test_itercols(self,table_types):
```
Comment (taldcroft) [inline astropy/table/tests/test_table.py]: PEP8 (space after comma)

File: astropy/table/tests/test_table.py [hunk 3]
```diff
@@ -333,6 +333,14 @@ def test_2(self, table_types):
...
+        t = table_types.Table()
```
Comment (taldcroft) [inline astropy/table/tests/test_table.py]: Not a big deal, but you can be more concise with:
```
names = ['a', 'b', 'c']
t = table_types.Table([[1], [2], [3]], names=names)
```

File: astropy/table/tests/test_table.py [hunk 4]
```diff
@@ -333,6 +333,14 @@ def test_2(self, table_types):
...
+        t.add_column(table_types.Column(name='a', data=[1, 2, 3]))
+        t.add_column(table_types.Column(name='b', data=[4, 5, 6]))
+        t.add_column(table_types.Column(name='c', data=[7, 8, 9]))
+        for col in t.itercols():
```
Comment (taldcroft) [inline astropy/table/tests/test_table.py]: Make this test be:
```
for name, col in izip(names, t.itercols()):
    assert name == col.name
    assert isinstance(col, table_types.Column)
```

**Comment by taldcroft** [comment]
@pulsestaysconstant - finished my review, this looks good apart from minor comments.  Please fix up and rebase and then we can merge!

**Comment by mwcraig** [comment]
@pulsestaysconstant -- looks like the remaining failures are because you need to import `izip` (it is part of `itertools`).

Try that change, then run `python setup.py test -P table` locally make sure the tests pass. Once they do, let me know and I'll walk you through an interactive rebase to squash all of the commits down to one.

**Comment by mwcraig** [comment]
@taldcroft -- ready to merge if it looks good to you. Had quite the git adventure with this one!

**Comment by taldcroft** [comment]
@mwcraig - code wise this looks OK.  I'm a little concerned about the astropy-helpers change and subsequent reversion that happens in the two commits.  Maybe a final squash is in order?  Hopefully this would not be much of an adventure.

**Comment by MSeifert04** [comment]
@mwcraig I think there is no `itertools.izip` in python 3.x anymore. I think that has to be loaded from six: http://pythonhosted.org/six/#module-six.moves

**Comment by mwcraig** [comment]
OK, let's wait to see what happens with the tests...

**Comment by taldcroft** [comment]
The import should be `from ...extern.six.moves import zip as izip`.  (Hopefully I have the right number of dots).  (the `import zip` or `import zip as izip` is a matter of preference here, either way is fine by me).

**Comment by mwcraig** [comment]
@taldcroft -- assuming this passes, I'll take care of the squash...

**Comment by taldcroft** [comment]
@mwcraig - thanks.  Tests are passing now.

**Comment by mwcraig** [comment]
The squash/rebase has been moved to #4888

**Pull request closed without merging**